app/src/main/java/com/example/bankdetails/utils/Utils.kt

```diff
@@ -22,3 +24,13 @@ fun Activity.hideSoftInput() {
 fun Fragment.hideSoftInput() = requireActivity().hideSoftInput()
 fun View.showSoftInput() = (context.getSystemService(Context.INPUT_METHOD_SERVICE)
         as InputMethodManager).showSoftInput(this, InputMethodManager.SHOW_FORCED)
+
+fun isValidIFSCode(str: String?): Boolean {
```

Comment:
In most cases (except callbacks or lamdas) it doesn't make sense to pass an optional to a function. A function takes an argument and returns an output. If the input isn't there it doesn't make sense to call the function in the first place.

Also not a big fan of `Utils` in general. What if you added this function as an extension on string? then you can use something like `"HDFC000".isValidIFSC()`

Reply:
Done